File: Dec2020/30Dec/SummaryRanges.cpp
/*
Problem name: 228. Summary Ranges
Problem link: https://leetcode.com/problems/summary-ranges/
*/

class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> result;
        int len=nums.size();
        for(int i=0;i<len;++i){
            int first=nums[i], flag=0;
            while(i+1 < len && nums[i]+1 == nums[i+1]){
                    i++;
                    flag=1;
            } 
            string temp="";
            if(flag == 1){                           
                temp = temp + to_string(first)+"->"+ to_string(nums[i]);
            }
            else{
                temp = temp + to_string(first);
            }
            result.push_back(temp);
        }
        return result;
    }
};

File: Dec2020/30Dec/SummaryRanges.java
/*
Problem Name: 228. Summary Ranges
Problem Link: https://leetcode.com/problems/summary-ranges/
*/
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> list = new ArrayList<String>();
        if(nums.length == 0)
            return list;
        int last = nums[0], n = nums.length;
        for(int i=1;i<n;i++){
            if(nums[i] != nums[i-1] + 1){
                if(last == nums[i-1])
                    list.add(last + "");
                else
                    list.add(last + "->" + nums[i-1]);
                last = nums[i];
            }
        }
        if(last == nums[n-1])
            list.add(last + "");
        else
            list.add(last + "->" + nums[n-1]);
        return list;
    }
}
